### ISSUE TITLE
Up arrow generates JS error

### ISSUE THREAD

**Issue by Karmalakas** [select2/select2#3523]
https://jsfiddle.net/urkaj76k/

Pressing Up arrow key, generates _Uncaught TypeError: Cannot read property 'top' of undefined_

**Comment by kevin-brown** [comment]
I was able to reproduce this, here's the traceback

```js
Uncaught TypeError: Cannot read property 'top' of undefined
(anonymous function) @ select2.full.js:1076
S2.define.Observable.invoke @ select2.full.js:614
S2.define.Observable.trigger @ select2.full.js:604
S2.define.Select2.trigger @ select2.full.js:5167
(anonymous function) @ select2.full.js:5100
S2.define.Observable.invoke @ select2.full.js:614
S2.define.Observable.trigger @ select2.full.js:604
S2.define.Select2.trigger @ select2.full.js:5167
(anonymous function) @ select2.full.js:5018
S2.define.Observable.invoke @ select2.full.js:614
S2.define.Observable.trigger @ select2.full.js:608
(anonymous function) @ select2.full.js:1820
jQuery.event.dispatch @ jquery-git.js:4979
jQuery.event.add.elemData.handle @ jquery-git.js:4659
```

The issue appears to be that hitting UP will trigger the highlighting routines to shift the highlight, but that can't be done because there are no options in the dropdown. This should only happen if there is nothing in the results box, which currently can only happen if there is an AJAX error. Otherwise it will display the normal "no results found" message.

I'm tempted to call this a duplicate of https://github.com/select2/select2/issues/3501.

**Comment by Karmalakas** [comment]
Actually I noticed this when I was trying to make options appear for multiple select, but was passing Ajax delay and dataType parameters. So no Ajax call was made, but also no options displayed :) That link to JsFiddle is what I mean

**Comment by raapperez** [comment]
This error still occurs in version 4.0.5.

**Comment by alexweissman** [comment]
Seems like it only happens in controls populated via AJAX, when no results are returned by a request (even a successful request).

**Comment by alexweissman** [comment]
Fixed in #5127.
